kmod/src/super.c

```diff
@@ -511,10 +511,10 @@ static int scoutfs_fill_super(struct super_block *sb, void *data, int silent)
 	mapping_set_gfp_mask(sb->s_bdev->bd_inode->i_mapping, GFP_NOFS);
 
 	sbi = kzalloc(sizeof(struct scoutfs_sb_info), GFP_KERNEL);
-	sb->s_fs_info = sbi;
```

Comment:
Careful.  that's a load bearing sb->s_fs_info = NULL.  It affects cleanup paths that look to see if s_fs_info was built up.  Leave it, and move the sb->s_fs_info below.